ksqldb-engine/src/main/java/io/confluent/ksql/function/udf/string/Concat.java

```diff
@@ -16,23 +16,31 @@
 package io.confluent.ksql.function.udf.string;
 
 import io.confluent.ksql.function.KsqlFunctionException;
-import io.confluent.ksql.function.udf.Kudf;
+import io.confluent.ksql.function.udf.Udf;
+import io.confluent.ksql.function.udf.UdfDescription;
+import io.confluent.ksql.function.udf.UdfParameter;
 import java.util.Arrays;
 import java.util.Objects;
 import java.util.stream.Collectors;
 
-public class ConcatKudf implements Kudf {
-  public static final String NAME = "CONCAT";
+@UdfDescription(
+    name = "concat",
+    description = "Concatenate an arbitrary number of string fields together")
+public class Concat {
 
-  @Override
-  public String evaluate(final Object... args) {
-    if (args.length < 2) {
-      throw new KsqlFunctionException(NAME + " should have at least two input argument.");
+  @Udf
+  public String concat(@UdfParameter(
+      description = "The varchar fields to concatenate") final String... inputs) {
+    if (inputs == null) {
+      return null;
+    }
+    if (inputs.length < 2) {
+      throw new KsqlFunctionException("Concat should have at least two input arguments.");
     }
```

Comment:
I know this isn't your original code, but does this limitation makes sense? I feel like `CONCAT('foo')` could reasonably return just `'foo'`. I can imagine passing in an array of values and expect it to work (now that we support variadic) even for the cases where there's fewer than 2 (or even if the array is empty, so `CONCAT([])` would return `""`)

Reply:
You make a fair point - and the same thought went through my head while going through the code but i decided not to disturb the status quo too much (for once!).Happy to change it though, i think what you are suggesting makes more intuitive sense, as it would bring the behavior into line with what happens if you pass more than 2 inputs but less than 2 of them are non-null - e.g. `concat(null, 'foo', null)` is already processed without complaint and returns `foo`.  WDYT ?

Reply:
+1 that makes a lot of sense (and in line with what we would do today even before this PR if we had `CONCAT(null, 'foo')`)

Reply:
Done!